tracpro/polls/models.py

```diff
@@ -772,6 +792,24 @@ def category_counts_by_pollrun(self):
         counts.sort(key=lambda (category, pollrun_counts): natural_sort_key(category))
         return counts
 
+    def get_answer_summaries_regions(self):
+        """Return the sum and average of numeric answers to each pollrun."""
+        """Split out values by regions"""
+        answer_sums_list = []
+        answer_avgs_list = []
+
+        for region in self.get_regions():
+            region_answers = self.filter(response__contact__region=region)
+            # Get the dictionaries of sums and averages per pollrun
+            answer_sums, answer_avgs = region_answers.get_answer_summaries()
```

Comment:
Any ideas on how to mitigate this off the top of your head? I'm not sure how to refactor this.